src/algokit/core/generate.py

```diff
@@ -0,0 +1,91 @@
+import dataclasses
+import logging
+from pathlib import Path
+
+import click
+
+from algokit.core.conf import ALGOKIT_CONFIG, get_algokit_config
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass(kw_only=True)
+class Generator:
+    name: str
+    path: str
+    description: str | None = None
+
+
+def _format_generator_name(name: str) -> str:
+    """
+    Format the generator name to be used as a command name.
+    :param name: Generator name.
+    :return: Formatted generator name.
+    """
+
+    return name.strip().replace(" ", "-").replace("_", "-")
+
+
+def run_generator(answers: dict, path: Path) -> None:
+    """
+    Run the custom generator with the given answers and path.
+    :param answers: Answers to pass to the generator.
+    :param path: Path to the generator.
+    """
+
+    # copier is lazy imported for two reasons
+    # 1. it is slow to import on first execution after installing
+    # 2. the import fails if git is not installed (which we check above)
+    from copier.main import Worker  # type: ignore[import]
```

Comment:
This (or perhaps in the cli layer) should check for the git command, and raise a helpful error if it isn't present. See init command here https://github.com/algorandfoundation/algokit-cli/blob/main/src/algokit/cli/init.py#L188

Reply:
@daniel-makerx added a check on cli side before running the run generator in core. And corresponding test let me know if there is anything else